power_grid_model_c/power_grid_model/include/power_grid_model/job_interface.hpp

```diff
@@ -18,20 +19,31 @@ template <typename Adapter> class JobInterface {
   public:
     // the multiple  NOSONARs are used to avoid the complaints about the unnamed concepts
     template <typename ResultDataset>
-    void calculate(ResultDataset const& result_data, Idx pos = 0)
+    void calculate(ResultDataset const& result_data, Idx pos, Logger& logger)
         requires requires(Adapter& adapter) { // NOSONAR
-            { adapter.calculate_impl(result_data, pos) } -> std::same_as<void>;
+            { adapter.calculate_impl(result_data, pos, logger) } -> std::same_as<void>;
         }
     {
-        return static_cast<Adapter*>(this)->calculate_impl(result_data, pos);
+        static_cast<Adapter*>(this)->calculate_impl(result_data, pos, logger);
+    }
+    template <typename ResultDataset> void calculate(ResultDataset const& result_data, Idx pos = 0) {
+        static constexpr NoLogger logger{};
```

Comment:
```suggestion
        static constexpr NoLogger no_logger{};
```
And, we can also have only one no_logger object and use it evereywhere, right?

Reply:
I would prefer to keep the `static constexpr NoLogger no_logger{}` local to avoid potential thread safety issues and so on. Right now it shouldn't be a problem, but it may be in the future.

To be honest, I actually think I should remove this overload alltogether and always request a logger to be explicitly passed in, let it be a `NoLogger` or an actual one. If the user is not interested in logging, we can just have a check for that and create a `NoLogger` explicitly.

The more I look at this I dislike it more. I'll remove these overloads and let's see how does it look like.

Reply:
See [c4e66a8](https://github.com/PowerGridModel/power-grid-model/pull/1122/commits/c4e66a8c7928da1560497605c411eda8f7941ca7), this should be better.